paimon-core/src/main/java/org/apache/paimon/index/PartitionIndex.java

```diff
@@ -137,4 +153,42 @@ public static PartitionIndex loadIndex(
         }
         return new PartitionIndex(mapBuilder.build(), buckets, targetBucketRowNumber);
     }
+
+    public static int[] getMaxBucketsPerAssigner(int maxBuckets, int assigners) {
```

Comment:
I cannot understand `getMaxBucketsPerAssigner` and `getSpecifiedMaxBuckets`. Why not just same to `maxBucketId < maxBucketsNum - 1`?